client/state/sites/reducer.js

```diff
@@ -216,6 +217,20 @@ export function items( state = null, action ) {
 				},
 			};
 		}
+
+		case EDITOR_TYPE_SET: {
+			const { siteId, editor } = action;
+			if ( state[ siteId ] && editor.match( /^(classic|gutenberg)$/ ) ) {
```

Comment:
if we need this validation we can keep it in but I don't think we need it. if it comes from the API then we should validate it at the boundary where it comes in. if we're guarding against code abuse internally then we're probably fine skipping the validation

Reply:
It was just nice to have validation, especially considering we know the only two values to accept. It wasn't a defence against any clear issue.